tests/Doctrine/Migrations/Tests/Metadata/Storage/TableMetadataStorageTest.php

```diff
@@ -355,4 +355,42 @@ public function testResetWithEmptySchema(): void
         );
         self::assertCount(0, $this->connection->fetchAllAssociative($sql));
     }
+
+    public function testGetSql(): void
+    {
+        $this->storage->ensureInitialized();
+
+        $result = new ExecutionResult(new Version('2230'), Direction::UP, new DateTimeImmutable('2010-01-05 10:30:21'));
+
+        $queries = $this->storage->getSql($result);
+
+        self::assertCount(2, $queries);
+        self::assertSame('-- Version 2230 update table metadata', $queries[0]->getStatement());
+        self::assertSame('INSERT INTO doctrine_migration_versions (version, executed_at, execution_time) VALUES (\'2230\', \'2010-01-05 10:30:21\', 0)', $queries[1]->getStatement());
+
+        foreach ($queries as $query) {
+            $this->connection->executeStatement($query);
+        }
+
+        $sql = sprintf(
+            'SELECT * FROM %s WHERE version = 2230',
+            $this->connection->getDatabasePlatform()->quoteIdentifier($this->config->getTableName())
+        );
+
+        self::assertCount(1, $this->connection->fetchAllAssociative($sql));
+
+        $result = new ExecutionResult(new Version('2230'), Direction::DOWN, new DateTimeImmutable('2010-01-05 10:30:21'));
+
+        $queries = $this->storage->getSql($result);
+
+        self::assertCount(2, $queries);
+        self::assertSame('-- Version 2230 update table metadata', $queries[0]->getStatement());
+        self::assertSame('DELETE FROM doctrine_migration_versions WHERE version = \'2230\'', $queries[1]->getStatement());
+
+        foreach ($queries as $query) {
+            $this->connection->executeStatement($query);
+        }
+
+        self::assertCount(0, $this->connection->fetchAllAssociative($sql));
```

Comment:
This is quite repetitive, please use a data provide to ensure both segment do use the same logic.

Reply:
Do you mean phpunit @dataProvider annotation? If so i disagree.

Queries in this test depends on each other: DOWN doesnt make sense without UP. I test exactly this situation - generate and execute UP queries, check that migration was inserted in metadata table and after that execute DOWN queries and check that migration was deleted from metadata table.

Reply:
similar behaviour presented in other test - https://github.com/doctrine/migrations/blob/3.5.x/tests/Doctrine/Migrations/Tests/Metadata/Storage/TableMetadataStorageTest.php#L339

Reply:
Ah I see, that makes sense.